src/app/file-manager/file-download-modal/file-download-modal.component.ts

```diff
@@ -11,38 +10,172 @@ import { DownloadValidatorService } from '@app/file-manager/services/download-va
 import { Type } from '@app/file-manager/models/type';
 import { Router } from '@angular/router';
 import { ItemStatus } from '@app/file-manager/models/item-status';
-import { TransferablesGridComponent} from '@app/file-manager/transferables-grid/transferables-grid.component';
+import { TransferablesGridComponent } from '@app/file-manager/transferables-grid/transferables-grid.component';
+import { BucketService } from '../services/bucket.service';
+import { FilterSizePipe } from '../filters/filesize-filter';
+
+import { Observable } from 'rxjs/Observable';
+import { BehaviorSubject } from 'rxjs/BehaviorSubject';
 
 @Component({
   selector: 'app-file-download-modal',
   templateUrl: './file-download-modal.component.html'
 })
-export class FileDownloadModalComponent  {
+export class FileDownloadModalComponent implements OnInit {
   @Output('done') done: EventEmitter<any> = new EventEmitter();
 
   preserveStructure = true;
   directory = 'Choose Directory...';
   isValid = false;
-  files: TreeNode[];
   msgs: Message[] = [];
   verify: DiskStatus;
+
+  selectedFiles: Item[] = [];
+
+  fileCount = 0;
+  totalSize = 0;
+  totalFiles = 0;
+
+  loadingFiles = false;
+  loadingFolders = 0;
   downloadFiles: Item[] = [];
+  filesMap: Map<String, Item>;
+
+  fileCountObservable: BehaviorSubject<number> = new BehaviorSubject(this.fileCount);
+  totalSizeObservable: BehaviorSubject<number> = new BehaviorSubject(this.totalSize);
+
+  myFiles: Item[] = [];
 
   constructor(
     private downloadValidator: DownloadValidatorService,
     private transferablesGridComponent: TransferablesGridComponent,
     private store: Store<AppState>,
+    private zone: NgZone,
     public dialogRef: MatDialogRef<FileDownloadModalComponent>,
     public router: Router,
+    public bucketService: BucketService,
     @Inject(MAT_DIALOG_DATA) public data: any) {
-      if (localStorage.getItem('directory') !== null) {
-        this.directory = localStorage.getItem('directory');
-        this.isValid = true;
+    this.filesMap = new Map();
+
+    if (localStorage.getItem('directory') !== null) {
```

Comment:
Rather than retrieving from `localStorage` multiple times, this could be phrased as:
```
var directory = localStorage.getItem('directory')
if (directory !== null) {
    this.directory = directory
    this.isValid = true;
}
```

Reply:
makes sense. Will fix it.